02-basics-2/40-marked-emails/MarkedEmailsApp.js

```diff
@@ -32,19 +32,21 @@ export const emails = [
 export default defineComponent({
   name: 'MarkedEmailsApp',
 
-  setup() {},
+  setup() {
+    const inputValue = ref('')
+    const isIncludes = email => email.includes(inputValue.value) && inputValue.value
```

Comment:
It isn't forbidden in general case and not a real issue in this component, but in general using methods in templates should be avoided. It makes it less declarative. And developers need to check if it has no side effect and performant.

Especially if it's not just an utility or formatting but data processing.

Reply:
Instead, we should prepare data using `computed`. As the result we need to have something, that can be used in a template directly without additional computation. Where in each item we know if it is marked.

Reply:
It's a lot harder than I thought